client/stripe-utils/utils.js

```diff
@@ -222,7 +222,12 @@ export const getPaymentMethodTypes = ( paymentMethodType = null ) => {
 	const paymentMethodsConfig = getStripeServerData()?.paymentMethodsConfig;
 
 	if ( paymentMethodType === null ) {
-		return Object.keys( paymentMethodsConfig || {} );
+		if ( getStripeServerData()?.isCheckout ) {
+			return Object.keys( paymentMethodsConfig || {} );
+		}
+
+		// If we're on the My Account > Add payment method page make sure we only support the card paymentMethodType.
+		return [ 'card' ];
```

Comment:
I made this change to fix an issue where the Stripe PE wouldn't load if I had no items in my cart and store currency set to EUR. Here's the console error I was getting:

![image](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/2275145/262d1e63-feab-4f02-bfd2-7598e531bcbc)

My understanding of the issue is `getPaymentMethodTypes()` was returning payment methods that don't support `mode` set to `'setup'` and amounts `0` and that's throwing the issue. Therefore to fix this, I made it so that if we're not on the checkout page, we return just ['card'] as the list available payment method types.

Once we implement Split PE I'm assuming we'll do something similar to WooPayments where any gateway that doesn't support the `add-payment-method` flag won't be loaded on the page to begin with.

Reply:
Sounds good, yup. If I recall correctly, SEPA was the only APM that could be added here, but I agree, we can check that when handling APMs 🙂